API.md

```diff
@@ -1,4 +1,111 @@
-# Authmosphere API documentation
+# Authmosphere API Documentation
+
+## OAuth Tooling
+
+This tooling provides helper functions to request and validate tokens based on the OAuth 2.0 [RFC 6749](https://tools.ietf.org/html/rfc6749) specification.
+
+### getAccessToken
+
+Requests a token based on the given configuration (which specifies the grant type and corresponding parameters). See the [`OAuthConfig` documentation](#tbd) for details.
+
+#### Usage
+
+```ts
+import {
+  OAuthGrantType,
+  ClientCredentialsGrantConfig,
+  Token,
+  getAccessToken
+} from 'authmosphere';
+
+const config: ClientCredentialsGrantConfig = {
+  grantType: OAuthGrantType.CLIENT_CREDENTIALS_GRANT,
+  credentialsDir: './crendentials',
+  accessTokenEndpoint: 'example.com/token',
+  scopes: ['my-app.read', 'my-app.write'];
+};
+
+getAccessToken(config)
+  .then((token: Token) => {
+    // ...use the token...
+  })
+  .catch((err) => {
+    // ...handle the error...
+  });
+```
+
+#### Signature
+
+`getAccessToken(config[, logger]): Promise<Token>`
+
+#### Arguments
+
+* [`config: OAuthConfig`](#tbd) - OAuth configuration for the request (specify grant type and corresponding parameters)
+* [`logger?: Logger`](#tbd) - Optional logger
+
+#### Returns
+
+`Promise<Token>`
+
+### getTokenInfo
+
+Requests validation information from the specified `tokenInfoUrl` and returns a `Promise` which resolves with these information, if the token is valid. Otherwise, it rejects with an error.
+
+#### Usage
+
+```ts
+import {
+  Token,
+  getTokenInfo
+} from 'authmosphere';
+
+getTokenInfo('example.com/tokeninfo', '1234-5678-9000')
+  .then((token: Token) => {
+    // ...token is valid...
+  })
+  .catch((err) => {
+    // ...token is invalid...
+  })
+```
+
+#### Signature
+
+`getTokenInfo<T>(tokenInfoUrl, accessToken[, logger]): Promise<Token<T>>`
+
+#### Arguments
+
+* `tokenInfoUrl: string` - OAuth endpoint for validating tokens
+* `accessToken: string` - Token to be validated
+* [`logger?: Logger`](#tbd) - Optional logger
```

Comment:
```
[`logger?: Logger`](./src/types/Logger.ts)
```

Reply:
Would rather reference the API doc section of the type, we wanted to add this right?

Reply:
Not sure if you can do anchor reference in markdown files

Reply:
At least it is possible to set anchor links to headlines :) 
https://stackoverflow.com/questions/6695439/how-to-link-to-a-named-anchor-in-multimarkdown

Need to stay constistent here, as I linked to the Types so far. But for logger I think a reference to the docu would be better.

Reply:
I think it is fine in general to link the types. Howerver, where we actually have a detailed doc we should link the doc. Probably this will be the case for `OAuthConfig`, `OAuthConfigType` (?) and `Logger`